src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/EntrypointSelectionStep.cs

```diff
@@ -79,6 +85,29 @@ protected UpgradeStepInitializeResult InitializeImpl(IUpgradeContext context)
                 return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected only project.", BuildBreakRisk.None);
             }
 
+            if (!string.IsNullOrEmpty(_entryPoint))
+            {
+                var entryPointProject = projects.Where(i => Path.GetFileName(i.FilePath) == _entryPoint).FirstOrDefault();
+                if (entryPointProject is not null)
+                {
+                    context.SetEntryPoint(entryPointProject);
+                    Logger.LogInformation("Setting entrypoint to user selected project in solution: {Project}", _entryPoint);
+
+                    return new UpgradeStepInitializeResult(UpgradeStepStatus.Complete, "Selected user's choice of entry point project.", BuildBreakRisk.None);
+                }
+                else
+                {
+                    Logger.LogInformation("Invalid Entry-Point Project specified : {Project}", _entryPoint);
+                }
+            }
+
+            if (context.EntryPoint is null && _isNonInteractiveMode)
```

Comment:
Instead of a flag like this, why not have `IUserInput` say whether it is interactive, ie `IUserInput.IsInteractive`?

Reply:
This is what I tried first, but it seems unnecessary in case of interactive mode as this is relevant only in the non-interactive mode.